LibCore/CProcessHelper.cpp

```diff
@@ -0,0 +1,107 @@
+#include "CProcessHelper.h"
+#include "CFile.h"
+
+#include <stdio.h>
+#include <pwd.h>
+
+CProcessHelper::CProcessHelper()
+{
+    setpwent();
+    while (auto* passwd = getpwent())
+	m_usernames.set(passwd->pw_uid, passwd->pw_name);
+    endpwent();
+}
+
+HashMap<pid_t, RetainPtr<CProcessInfo>> CProcessHelper::get_map()
+{
+    HashMap<pid_t, RetainPtr<CProcessInfo>> res;
+    int error = 0;
+  
+    if ((error = update_map(res)) != 0) {
+	fprintf(stderr, "CProcessHelper::update_map failed with code %d\n", error);
+    }
+
+    return res;
+}
+
+int CProcessHelper::update_map(HashMap<pid_t, RetainPtr<CProcessInfo>>& map)
+{
+    CFile file("/proc/all");
+    if (!file.open(CIODevice::ReadOnly)) {
+	fprintf(stderr, "CProcessHelper : failed to open /proc/all: %s\n", file.error_string());
+	return 1;
+    }
+
+    for (;;) {
+	auto line = file.read_line(1024);
+
+	if (line.is_empty())
+	    break;
+
+	auto chomped = String((const char*)line.pointer(), line.size() - 1, Chomp);
+	auto parts = chomped.split_view(',');
+
+	if (parts.size() < 18)
+	    break;
+
+	bool ok = false;
+	RetainPtr<CProcessInfo> process = RetainPtr<CProcessInfo>(new CProcessInfo);
```

Comment:
This is a subtle bug, but this CProcessInfo would actually leak.
All retainable object begin with retain_count==1, and the first owner must adopt it like so:
RetainPtr<CProcessInfo> process = adopt(*new CProcessInfo);
If you do this:
RetainPtr<CProcessInfo> process = RetainPtr<CProcessInfo>(new CProcessInfo);
The retain_count will now be 2.

It would be good to have a make_retained() helper that prevents this bug but I haven't been able to solve it perfectly yet..